src/terrain/KM_Terrain.pas

```diff
@@ -16,6 +16,27 @@ interface
                     vu_NESW);   //Vertex is used NE-SW like this: /
   TFenceType = (fncNone, fncCorn, fncWine, fncHousePlan, fncHouseFence);
 
+  TKMTileChangeType = (tctTerrain, tctHeight, tctObject);
+
+  TKMTileChangeTypeSet = set of TKMTileChangeType;
+
+  TKMTerrainTileBrief = packed record

```

Comment:
It's less efficient to use packed records unless we care about padding for saving/loading (or RAM usage in really big arrays). Packing forces the CPU to do more work extracting each element since they are not nicely aligned. None of these records need packing as far as I can see.